source/common/runtime/runtime_impl.cc

```diff
@@ -23,60 +23,78 @@ namespace Runtime {
 
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
+void RandomGeneratorImpl::random_bytes(uint8_t* out, size_t out_len) {
+  if (out_len >= 1024) {
+    int rc = RAND_bytes(out, out_len);
+    ASSERT(rc == 1);
+    UNREFERENCED_PARAMETER(rc);
+    return;
+  }
+
+  static thread_local uint8_t buffered[2048];
+  static thread_local size_t buffered_idx = sizeof(buffered);
+
+  do {
+    if (buffered_idx >= sizeof(buffered)) {
+      int rc = RAND_bytes(buffered, sizeof(buffered));
+      ASSERT(rc == 1);
+      UNREFERENCED_PARAMETER(rc);
+      buffered_idx = 0;
+    }
+
+    size_t available = std::min(sizeof(buffered) - buffered_idx, out_len);
+
+    std::memcpy(out, reinterpret_cast<void*>(&buffered[buffered_idx]), available);
+
+    buffered_idx += available;
+    out_len -= available;
+    out += available;
+  } while (out_len);
+}
+
 std::string RandomGeneratorImpl::uuid() {
   // Create UUID from Truly Random or Pseudo-Random Numbers.
   // See: https://tools.ietf.org/html/rfc4122#section-4.4
   uint8_t rand[16];
-  int rc = RAND_bytes(rand, 16);
-  ASSERT(rc == 1);
-  UNREFERENCED_PARAMETER(rc);
+  random_bytes(rand, 16);
   rand[6] = (rand[6] & 0x0f) | 0x40; // UUID version 4 (random)
   rand[8] = (rand[8] & 0x3f) | 0x80; // UUID variant 1 (RFC4122)
 
   // Convert UUID to a string representation, e.g. a121e9e1-feae-4136-9e0e-6fac343d56c9.
   static const char* const hex = "0123456789abcdef";
-  std::string uuid;
-  uuid.reserve(UUID_LENGTH);
+  static thread_local char uuid[] = "------------------------------------";
```

Comment:
There was a small performance gain (~5% in benchmark, AFAIR) because of that, since we don't have to write `-` into 4 places each time when using `static`.

...which reminds me that we could hardcode `4` here as well, for another slight bump.